endpoints/data_objects/src/main.cpp

```diff
@@ -1375,42 +1623,48 @@ namespace
 					return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
 				}
 
-				auto conn = irods::get_connection(client_info.username);
-
-				if (!fs::client::is_data_object(conn, lpath_iter->second)) {
-					return _sess_ptr->send(irods::http::fail(
-						res,
-						http::status::bad_request,
-						json{{"irods_response", {{"status_code", NOT_A_DATA_OBJECT}}}}.dump()));
+				const auto catalog_only_iter = _args.find("catalog-only");
+				if (catalog_only_iter == std::end(_args)) {
+					log::error("{}: Missing [catalog-only] parameter.", fn);
+					return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
 				}
 
-				fs::extended_remove_options opts{};
+				DataObjInp input{};
+				irods::at_scope_exit free_memory{[&input] { clearKeyVal(&input.condInput); }};
+
+				irods::strcpy_null_terminated(input.objPath, lpath_iter->second.c_str());
+
+				if (catalog_only_iter->second == "1") {
+					input.oprType = UNREG_OPR;
 
-				if (const auto iter = _args.find("no-trash"); iter != std::end(_args) && iter->second == "1") {
-					opts.no_trash = true;
+					if (const auto iter = _args.find("no-trash"); iter != std::end(_args) && iter->second == "1") {
+						log::error("{}: [catalog-only] and [no-trash] parameters are incompatible.", fn);
+						return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+					}
+				}
+				else if (const auto iter = _args.find("no-trash"); iter != std::end(_args) && iter->second == "1") {
+					addKeyVal(&input.condInput, FORCE_FLAG_KW, "");
 				}
 
-				if (const auto iter = _args.find("unregister"); iter != std::end(_args) && iter->second == "1") {
-					opts.unregister = true;
+				if (const auto iter = _args.find("admin"); iter != std::end(_args) && iter->second == "1") {
+					addKeyVal(&input.condInput, ADMIN_KW, "");
 				}
 
-				// There's no admin flag for removal.
-				fs::client::remove(conn, lpath_iter->second, opts);
+				auto conn = irods::get_connection(client_info.username);
+				const auto ec = rcDataObjUnlink(static_cast<RcComm*>(conn), &input);
```

Comment:
Does this suggest that `irods::experimental::filesystem` needs to grow some more options in order to be used for cases like this, or is it more that the interface doesn't support this case because it's not part of the `std::filesystem` interface with which developers are familiar?

Reply:
It does suggest that.

Will consider opening an issue for that.

Reply:
Seems reasonable

Reply:
Ok. Marking as resolved.